common/src/main/scala/akkapi/cluster/LedStripDriver.scala

```diff
@@ -58,33 +71,51 @@ class LedStripDriver private (context: ActorContext[LedStripDriver.Command],
   timers.startTimerAtFixedRate(WeaklyUpHeartBeat, weaklyUpIndicatorInterval)
 
   // TODO: Need to add customer supervision to handle IllegalArgumentException being thrown when a non-existing LED strip type is passed in
-  def run(hearbeatLedIsOn: Boolean, weaklyUpLedIsOn: Boolean, weaklyUpNodes: Set[Int]): Behavior[Command] = Behaviors.receiveMessage[Command] {
+  def run(hearbeatLedIsOn: Boolean,
+          weaklyUpLedIsOn: Boolean,
+          weaklyUpNodes: Set[Int],
+          blinkers: Map[String, ActorRef[LedPulser.Command]]): Behavior[Command] = Behaviors.receiveMessage[Command] {
+
+    case FlashLed(id, ledNumber, color, flashDuration, overRunColor) =>
+      val blinkersUpdated = if (! blinkers.contains(id)) {
+        val newBlinker = createLedPulser(id)
+        blinkers + (id -> newBlinker)
+      } else blinkers
+      blinkersUpdated(id) ! LedPulser.PulseLed(ledNumber, color, flashDuration, None)
+      run(hearbeatLedIsOn, weaklyUpLedIsOn, weaklyUpNodes, blinkersUpdated)
+
     case SetLedState(ledId, color, None) =>
       setPixelColorAndShow(ledId, color)
-      run(hearbeatLedIsOn, weaklyUpLedIsOn, weaklyUpNodes - ledId)
+      run(hearbeatLedIsOn, weaklyUpLedIsOn, weaklyUpNodes - ledId, blinkers)
     case SetLedState(ledId, color, Some(WeaklyUpBlinker)) =>
       setPixelColorAndShow(ledId, color)
-      run(hearbeatLedIsOn, weaklyUpLedIsOn, weaklyUpNodes + ledId)
+      run(hearbeatLedIsOn, weaklyUpLedIsOn, weaklyUpNodes + ledId, blinkers)
 
     case Heartbeat if hearbeatLedIsOn =>
       setPixelColorAndShow(HeartbeatLedNumber, Black)
-      run(hearbeatLedIsOn = false, weaklyUpLedIsOn, weaklyUpNodes)
+      run(hearbeatLedIsOn = false, weaklyUpLedIsOn, weaklyUpNodes, blinkers)
     case Heartbeat =>
       setPixelColorAndShow(HeartbeatLedNumber, heartbeatIndicatorColor)
-      run(hearbeatLedIsOn = true, weaklyUpLedIsOn, weaklyUpNodes)
+      run(hearbeatLedIsOn = true, weaklyUpLedIsOn, weaklyUpNodes, blinkers)
 
     case WeaklyUpHeartBeat if weaklyUpLedIsOn =>
       for (ledId <- weaklyUpNodes) setPixelColorAndShow(ledId, Black)
-      run(hearbeatLedIsOn, weaklyUpLedIsOn = false, weaklyUpNodes)
+      run(hearbeatLedIsOn, weaklyUpLedIsOn = false, weaklyUpNodes, blinkers)
     case WeaklyUpHeartBeat =>
       for (ledId <- weaklyUpNodes) setPixelColorAndShow(ledId, nodeWeaklyUpColor)
-      run(hearbeatLedIsOn, weaklyUpLedIsOn = true, weaklyUpNodes)
+      run(hearbeatLedIsOn, weaklyUpLedIsOn = true, weaklyUpNodes, blinkers)
   }.receiveSignal {
     case (_, signal) if signal == PostStop =>
       resetAllLeds(strip)
       Behaviors.same
   }
 
+  private def createLedPulser(id: String): ActorRef[LedPulser.Command] = {
+    val blinker = context.spawnAnonymous(LedPulser(settings, context.self))
+    context.watchWith(blinker, BlinkerTerminated(id))
```

Comment:
Maybe I'm missing something, but is `BlinkerTerminated` handled anywhere?

Reply:
Good catch. Adding the handling of this message.
More in general, this is part of a bigger context where a pulser could be stopped at some point in time (either autonomously based on some condition or controlled by the `LEDDriver` by sending some message). Right now, this isn't the case. Once started, a pulser stays alive until the application itself is stopped.